### PR TITLE
feat: enhance the filtering API to support device full-text search

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: title
Ticket: MEN-5527

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
Failure in `tests/test_mender_gateway.py::TestMenderGatewayOpenSource::test_deployment_two_devices_update_both::setup` is not relevant.

File: utils/search.go [hunk 1]
```diff
@@ -24,3 +24,20 @@ func ContainsString(val string, vals []string) bool {
...
 	return false
 }
+
+func TextToKeywords(in string) string {
+	s := []byte(in)
+	j := 0
+	for _, b := range s {
+		if ('a' <= b && b <= 'z') ||
+			('A' <= b && b <= 'Z') ||
+			('0' <= b && b <= '9') {
```
Comment (alfrunes) [inline utils/search.go:34]: Should we apply more general unicode rules? Or are we restricting values to ascii?
```suggestion
	for _, b := range in {
		if unicode.IsLetter(b) || unicode.IsDigit(b) {
```

File: utils/search.go [hunk 2]
```diff
@@ -24,3 +24,20 @@ func ContainsString(val string, vals []string) bool {
...
+			s[j] = b
+			j++
+		} else {
+			s[j] = ' '
+			j++
+		}
```
Comment (alfrunes) [inline utils/search.go:40]: Should we also trim consecutive white spaces?

**Review by alfrunes — APPROVED** [review]
The PR looks good for a PoC. I only have a couple of nitpicks and questions.
Should we also target a feature-branch for this until we at least have an assessment of the (write-) performance and other implications this feature might have?

File: utils/search.go (hunk 1 above)
Comment (tranchitella) [reply utils/search.go:34]: unicode is fine, thanks

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline

**Comment by tranchitella** [comment]
Again, failure in `tests/test_mender_gateway.py::TestMenderGatewayOpenSource::test_deployment_two_devices_update_both` is not relevant.

**Pull request merged**